fdbclient/ServerKnobs.cpp

```diff
@@ -421,6 +421,8 @@ void ServerKnobs::initialize(Randomize randomize, ClientKnobs* clientKnobs, IsSi
 	// Enable this knob only for experminatal purpose, never enable this in production.
 	// If enabled, all the committed in-memory memtable writes are lost on a crash.
 	init( ROCKSDB_DISABLE_WAL_EXPERIMENTAL,                    false );
+	init( ROCKSDB_SINGLEKEY_DELETES_ON_CLEARRANGE,             false );
```

Comment:
When this is set to true, it's probably better to set `ENABLE_CLEAR_RANGE_EAGER_READS` to false. I've noticed that a single key deletion will be converted to a range clear (via the eager reads), then RocksDB engine will unnecessarily do a range read later in this PR, find nothing, and adds a single key delete.

Reply:
@jzhou77 Got it! I added two knobs like this for now.
init( ROCKSDB_SINGLEKEY_DELETES_ON_CLEARRANGE,             false );
init( ENABLE_CLEAR_RANGE_EAGER_READS, !ROCKSDB_SINGLEKEY_DELETES_ON_CLEARRANGE );

If we manually override one of the knobs we lose this relationship. Any suggestions for a better way?

Reply:
Is it well tested disabling the EAGER_READ_CLEAR_RANGE?

Reply:
disabling the EAGER_READ_CLEAR_RANGE has performance issue with sqlite. not sure if rocksdb is the same. we have to test the combination. I am fine leave the knob as is until we figure out experimentally the settings.

Reply:
sure.. May be I will leave them independent and will currently set EAGER_READ_CLEAR_RANGE=true and ROCKSDB_SINGLEKEY_DELETES_ON_CLEARRANGE=false. Will add a comment that these knobs have contrary  functionalities.
Will experiment with these knobs in AWS and ckperf clusters and set them to the right values later. Thanks!